### ISSUE TITLE
Elternaccounts?

### ISSUE THREAD

**Issue by alessioC42** [alessioC42/lanis-mobile#107]
Eltern können Anscheinend auch Accounts habnen. 

Unter den Schüleraccounts im bereich "Benutzerdaten" gibt es auch ein Feld "Zugeordnete Eltern/Erziehungsberechtigte"

![image](https://github.com/alessioC42/lanis-mobile/assets/84250128/b1ccaa69-c594-44e8-b521-7bf9ad7adff8)

https://support.schulportal.hessen.de/knowledgebase.php?article=606
https://support.schulportal.hessen.de/knowledgebase.php?search=Eltern

Es stellt sich die Frage, wie ein Elternaccount mit "Mein Unterricht" (und evtl. "Nachrichten") umgeht. Es wäre möglich, dass das der Elternaccount zugriff auf die Ansichten des Schülers hat.

**Comment by CodeSpoof** [comment]
Eltern-Accounts an unserer Schule (5174) haben keinen Zugriff auf "Mein Unterricht". Nachrichten werden nur unterstützt an bestimmte Personen. So können Eltern nur Lehrkräfte anschreiben und auch nur die, die deren Kinder in einem Fach unterrichten.

**Comment by alessioC42** [comment]
Hey, hast du discord, das würde das jetzt einfacher machen?

**Comment by CodeSpoof** [comment]
ja, heiße codespoof

**Comment by CodeSpoof** [comment]
Nachrichten funktionieren einwandfrei, Vertretungsplan und Kalender auch.
"Mein Unterricht" ist jetzt doch verfügbar, sollte also freigeschaltet werden. Hattet ihr mit Lehrer-Accounts schon getestet und gab es Fehler mit dem "Mein Unterricht" Modul @alessioC42? sonst hätte ich die Sperre dabei einfach aufgehoben.

**Comment by alessioC42** [comment]
Ja, die Lehreraccounts sind da problematisch. Sie haben eben nicht die Sicht der Kursteilnehmer, sondern die Sicht des Kursadministrators -> HTML wird anders gerendert und Features werden einfach nicht unterstützt. 

Das ist einer der Gründe, warum wir die Sperre überhaupt eingeführt haben. Ich habe das auch mit 2 verschiedenen Lehreraccounts getestet.

**Comment by CodeSpoof** [comment]
Ich wüsste jetzt nicht, wie man die sonst auseinander halten könnte.

**Comment by alessioC42** [comment]
Bei den Benutzerdaten müsste zugeordneter Schüler oder sowas stehen, oder?

**Comment by CodeSpoof** [comment]
Schon, aber nicht immer leider. Das Schulportal ist da nicht sehr konsequent. Bei meiner Mutter nein, meinem Vater ja.

**Comment by alessioC42** [comment]
Und beide können auf "mein unterricht" zugreifen? Nicht nur deine Mutter?

**Comment by CodeSpoof** [comment]
Ja

**Comment by alessioC42** [comment]
Das ist ja so dumm. 

Dann würde ich beim ersten einloggen wenn ein nicht-schüleraccount festgestellt wird durch aufrufen von "mein Unterricht" herausfinden, ob es ein Elternaccount ist. Aber ich denke jetzt sollte erstmal das cleanup und das refactoring passieren. Danach kann das gemacht werden. 

v2.12.0 wird umfangreich

**Comment by alessioC42** [comment]
fixed by #327
